tools/patch/grammar.patch

```diff
@@ -0,0 +1,1005 @@
++    | 'break' { new astnodes.Break(...EXTRA) }
++    | 'continue' { new astnodes.Continue(...EXTRA) }
+     | &'global' global_stmt
+     | &'nonlocal' nonlocal_stmt
+-compound_stmt[stmt_ty]:
++compound_stmt[stmt]:
+     | &('def' | '@' | ASYNC) function_def
+     | &'if' if_stmt
+     | &('class' | '@') class_def
+@@ -81,182 +81,180 @@ compound_stmt[stmt_ty]:
+     | &'while' while_stmt
+ 
+ # NOTE: annotated_rhs may start with 'yield'; yield_expr must start with 'yield'
+-assignment[stmt_ty]:
++assignment[stmt]:
+     | a=NAME ':' b=expression c=['=' d=annotated_rhs { d }] {
+         CHECK_VERSION(
+             6,
+             "Variable annotation syntax is",
+-            _Py_AnnAssign(CHECK(_PyPegen_set_expr_context(p, a, Store)), b, c, 1, EXTRA)
++            new astnodes.AnnAssign(CHECK(pegen.set_expr_context(this, a, new astnodes.Store())), b, c, 1, ...EXTRA)
+         ) }
+     | a=('(' b=single_target ')' { b }
+          | single_subscript_attribute_target) ':' b=expression c=['=' d=annotated_rhs { d }] {
+-        CHECK_VERSION(6, "Variable annotations syntax is", _Py_AnnAssign(a, b, c, 0, EXTRA)) }
++        CHECK_VERSION(6, "Variable annotations syntax is", new astnodes.AnnAssign(a, b, c, 0, ...EXTRA)) }
+     | a=(z=star_targets '=' { z })+ b=(yield_expr | star_expressions) !'=' tc=[TYPE_COMMENT] {
+-         _Py_Assign(a, b, NEW_TYPE_COMMENT(p, tc), EXTRA) }
++         new astnodes.Assign(a, b, pegen.NEW_TYPE_COMMENT(this, tc), ...EXTRA) }
+     | a=single_target b=augassign ~ c=(yield_expr | star_expressions) {
+-         _Py_AugAssign(a, b->kind, c, EXTRA) }
++         new astnodes.AugAssign(a, b.kind, c, ...EXTRA) }
+     | invalid_assignment
+ 
+-augassign[AugOperator*]:
+-    | '+=' { _PyPegen_augoperator(p, Add) }
+-    | '-=' { _PyPegen_augoperator(p, Sub) }
+-    | '*=' { _PyPegen_augoperator(p, Mult) }
+-    | '@=' { CHECK_VERSION(5, "The '@' operator is", _PyPegen_augoperator(p, MatMult)) }
+-    | '/=' { _PyPegen_augoperator(p, Div) }
+-    | '%=' { _PyPegen_augoperator(p, Mod) }
+-    | '&=' { _PyPegen_augoperator(p, BitAnd) }
+-    | '|=' { _PyPegen_augoperator(p, BitOr) }
+-    | '^=' { _PyPegen_augoperator(p, BitXor) }
+-    | '<<=' { _PyPegen_augoperator(p, LShift) }
+-    | '>>=' { _PyPegen_augoperator(p, RShift) }
+-    | '**=' { _PyPegen_augoperator(p, Pow) }
+-    | '//=' { _PyPegen_augoperator(p, FloorDiv) }
+-
+-global_stmt[stmt_ty]: 'global' a=','.NAME+ {
+-    _Py_Global(CHECK(_PyPegen_map_names_to_ids(p, a)), EXTRA) }
+-nonlocal_stmt[stmt_ty]: 'nonlocal' a=','.NAME+ {
+-    _Py_Nonlocal(CHECK(_PyPegen_map_names_to_ids(p, a)), EXTRA) }
+-
+-yield_stmt[stmt_ty]: y=yield_expr { _Py_Expr(y, EXTRA) }
+-
+-assert_stmt[stmt_ty]: 'assert' a=expression b=[',' z=expression { z }] { _Py_Assert(a, b, EXTRA) }
+-
+-del_stmt[stmt_ty]:
+-    | 'del' a=del_targets &(';' | NEWLINE) { _Py_Delete(a, EXTRA) }
++augassign[AugOperator]:
++    | '+=' { pegen.augoperator(this, new astnodes.Add()) }
```

Comment:
could these be stored in consts instead of `new`-ing them every time?

Reply:
yeah that's a good idea. https://github.com/skulpt/skulpt_parser/issues/13